app/controllers/api/v1/users_controller.rb

```diff
@@ -0,0 +1,21 @@
+module Api
+  module V1
+    class UsersController < Api::V1::ApplicationController
+      def index
+        @users = if params[:provider].present?
+                   if params[:provider] == 'native'
```

Comment:
add before action for validation params[:provider], it must be only  native or oktaauth, if false render error(bad_request)

Reply:
Also move this "if" to private method

Reply:
```
def filtered_users
   params[:provider] == 'native' ? User.native_auth :  User.okta_auth
end
  
def obtain_users
  @users =  params[:provider].present? ?  filtered_users :  User.all
 end
```